programs/cw-ics08-wasm-eth/src/sudo.rs

```diff
@@ -159,11 +173,112 @@ mod tests {
     use cosmwasm_std::{
         coins, from_json,
         testing::{message_info, mock_env},
-        Binary,
+        Binary, Timestamp,
+    };
+    use ethereum_light_client::{
+        header::Header,
+        test_utils::fixtures::{self, InitialState, RelayerMessages, StepsFixture},
+    };
+    use ibc_proto::ibc::lightclients::wasm::v1::ClientMessage;
+    use prost::Message;
+
+    use crate::{
+        contract::instantiate,
+        msg::{VerifyClientMessageMsg, UpdateStateMsg},
+        test::mk_deps,
+        query::verify_client_message,
     };
-    use ethereum_light_client::test_utils::fixtures::{self, InitialState, StepsFixture};
 
-    use crate::{contract::instantiate, test::mk_deps};
+    use super::update_state;
+
+    #[test]
+    fn test_update_state() {
+        let mut deps = mk_deps();
+        let creator = deps.api.addr_make("creator");
+        let info = message_info(&creator, &coins(1, "uatom"));
+
+        let fixture: StepsFixture =
+            fixtures::load("TestICS20TransferNativeCosmosCoinsToEthereumAndBack_Groth16");
+
+        let initial_state: InitialState = fixture.get_data_at_step(0);
+
+        let client_state = initial_state.client_state;
+        let consensus_state = initial_state.consensus_state;
+
+        let client_state_bz: Vec<u8> = serde_json::to_vec(&client_state).unwrap();
+        let consensus_state_bz: Vec<u8> = serde_json::to_vec(&consensus_state).unwrap();
+
+        let msg = crate::msg::InstantiateMsg {
+            client_state: Binary::from(client_state_bz),
+            consensus_state: Binary::from(consensus_state_bz),
+            checksum: b"checksum".into(),
+        };
+        instantiate(deps.as_mut(), mock_env(), info, msg).unwrap();
+
+        let relayer_messages: RelayerMessages = fixture.get_data_at_step(1);
+        let (update_client_msgs, _, _) = relayer_messages.get_sdk_msgs();
+        assert!(!update_client_msgs.is_empty());
+        let headers = update_client_msgs
+            .iter()
+            .map(|msg| {
+                let client_msg =
+                    ClientMessage::decode(msg.client_message.clone().unwrap().value.as_slice())
+                        .unwrap();
+                serde_json::from_slice(client_msg.data.as_slice()).unwrap()
+            })
+            .collect::<Vec<Header>>();
+
+        let header = headers[0].clone();
+
+        let header_bz: Vec<u8> = serde_json::to_vec(&header).unwrap();
+        let header_bz2: Vec<u8> = header_bz.clone();
+        let header_bz3: Vec<u8> = header_bz.clone();
```

Comment:
did this just to shut the compiler up but lmk what the canonical solution is @srdtrk